src/components/FormOutcomeList/index.js

```diff
@@ -7,43 +7,59 @@ import { COLOR_SCHEME_DEFAULT } from 'utils/constants'
 
 import './formOutcomeList.less'
 
-const FormOutcomeList = ({ fields, label, meta: { error, invalid } }) => (
-  <div className="formOutcomeList">
-    <label htmlFor="outcomes" className="formOutcomeList__label">
-      {label}
-    </label>
-    {fields.map((field, index) => (
-      <div key={index} className={'formOutcomeList__entry'}>
-        <div className={'entry__color'} style={{ backgroundColor: COLOR_SCHEME_DEFAULT[index] }} />
-        <Field
-          component={FormInput}
-          name={`${field}`}
-          onChange={(e, val) => {
-            if (index === fields.length - 1 && val != null && val.length > 0) {
-              fields.push('')
-            }
-          }}
-          className="formOutcomeListInput"
-          placeholder="Add another..."
-        />
-        {fields.length > 2 && (
-          <a
-            className="entry__delete"
-            href=""
-            tabIndex="-1"
-            onClick={(e) => {
-              e.preventDefault()
-              fields.remove(index)
-            }}
-          >
-            Delete
-          </a>
-        )}
+class FormOutcomeList extends Component {
+  addOutcome = () => this.props.fields.push('')
+
+  addOutcomeOnChange = (event) => {
+    const { fields } = this.props
+    const { target } = event
+    const { value } = target
+    const index = +target.getAttribute('data-index')
+    if (index === fields.length - 1 && value != null && value.length > 0) {
+      fields.push('')
+    }
+  }
+
+  removeOutcome = (event) => {
+    event.preventDefault()
+    const { fields } = this.props
+    const index = event.target.getAttribute('data-index')
+    fields.remove(index)
+  }
+
+  render() {
+    const { fields, label, meta: { error, invalid } } = this.props
+    return (
+      <div className="formOutcomeList">
+        <label htmlFor="outcomes" className="formOutcomeList__label">
+          {label}
+        </label>
+        {fields.map((field, index) => (
+          <div key={index} className={'formOutcomeList__entry'}>
+            <div className={'entry__color'} style={{ backgroundColor: COLOR_SCHEME_DEFAULT[index] }} />
```

Comment:
Extract the style object into a constant. Otherwise the diff's algorithm always will identify this component as something that should be re-rendered.